CMakeLists.txt

```diff
@@ -0,0 +1,155 @@
+#Baysen Vision Inc. Structured Light project setup file
+#Copyright Reserved. 2013-2016
+
+cmake_minimum_required(VERSION 2.8.6 FATAL_ERROR)
+include(GenerateExportHeader)
+set(CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+project(slstudio)
+find_package(OpenCV REQUIRED)
+if(OpenCV_VERSION VERSION_LESS "3")
+    add_definitions(-DOPENCV2)
+endif(OpenCV_VERSION VERSION_LESS "3")
+
+find_package(Glew REQUIRED)
+find_package(OpenGL REQUIRED)
+
+find_package(PCL)
+find_package(VTK)
+find_package(Boost 1.47 COMPONENTS chrono filesystem system timer REQUIRED)
+
+set(USE_QT5 true)
+if(VTK_FOUND)
+    if(VTK_QT_VERSION VERSION_LESS "5")
+        set(USE_QT5 false)
+    endif(VTK_QT_VERSION VERSION_LESS "5")
+endif(VTK_FOUND)
+
+if(${USE_QT5})
+    find_package(Qt5 COMPONENTS Core Gui Test Widgets OpenGL)
+    # Instruct CMake to run moc automatically when needed.
+    set(CMAKE_AUTOMOC ON)
+else()
+    #Legacy qt4 support
+    find_package(Qt4 REQUIRED QtCore QtGui QtTest QtOpenGL)
+    include(${QT_USE_FILE})
+endif()
+
+include_directories(${CMAKE_CURRENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/src ${CMAKE_CURRENT_BINARY_DIR} 
+                    ${CMAKE_CURRENT_BINARY_DIR}/src ${CMAKE_CURRENT_BINARY_DIR}/src/camera
+                    ${CMAKE_CURRENT_BINARY_DIR}/src/projector
+                    ${OpenCV_INCLUDE_DIRS} ${Boost_INCLUDE_DIR} 
+                    ${GLEW_INCLUDE_DIR} ${OPENGL_INCLUDE_DIR})
+
+include_directories(${PROJECT_SOURCE_DIR}/src/camera)
+include_directories(${PROJECT_SOURCE_DIR}/src/projector)
+include_directories(${PROJECT_SOURCE_DIR}/src/codec)
+include_directories(${PROJECT_SOURCE_DIR}/src/triangulator)
+include_directories(${PROJECT_SOURCE_DIR}/src/calibrator)
+include_directories(${PROJECT_SOURCE_DIR}/src/tracker)
+
+add_subdirectory(${PROJECT_SOURCE_DIR}/src)
+add_subdirectory(${PROJECT_SOURCE_DIR}/src/camera)
+add_subdirectory(${PROJECT_SOURCE_DIR}/src/projector)
+
+#generate cmake config header file
+configure_file(${PROJECT_SOURCE_DIR}/slconfig.h.in
+        ${PROJECT_BINARY_DIR}/slconfig.h)
+
+#Set source file set
+file(GLOB UI_FILES src/SLStudio.ui 
```

Comment:
File glob with listing all the files IMHO does not make much sense. It is for collecting a set of files based on a pattern, so you can i.e.
`file(GLOB UI_FILES src/*.ui)
`
or say explicitly say:
```
set(UI_FILES src/SLStudio.ui 
   src/SLPreferenceDialog.ui
   ... 
)
```